app/models/question_answer_comment.rb

```diff
@@ -0,0 +1,56 @@
+# frozen_string_literal: true
+
+class QuestionAnswerComment < ActiveRecord::Base
+  include Trashable
+
+  # Bump this when changing MARKDOWN_FEATURES or MARKDOWN_IT_RULES
+  COOKED_VERSION = 1
+
+  belongs_to :topic
+  belongs_to :post
+  belongs_to :user
+
+  validates :topic_id, presence: true
+  validates :post_id, presence: true
+  validates :user_id, presence: true
+  validates :raw, presence: true
+  validates :cooked, presence: true
+  validates :cooked_version, presence: true
+
+  validate :ensure_can_comment, on: [:create]
+  before_validation :cook_raw, if: :will_save_change_to_raw?
+
+  MARKDOWN_FEATURES = %w{
+    censored
+    emoji
+  }
+
+  MARKDOWN_IT_RULES = %w{
+    emphasis
+    backticks
+    linkify
+    link
+  }
+
+  def self.cook(raw)
+    raw.gsub!(/(\n)+/, " ")
+    PrettyText.cook(raw, features_override: MARKDOWN_FEATURES, markdown_it_rules: MARKDOWN_IT_RULES)
+  end
+
+  private
+
+  def cook_raw
+    self.cooked = self.class.cook(self.raw)
+    self.cooked_version = COOKED_VERSION
```

Comment:
I'm setting the version but we don't automatically rebake yet. Will work on this in a future PR.